src/cookbooks/clients_last_seen_bits.md

```diff
@@ -0,0 +1,375 @@
+
+For retention calculations, we use forward-looking windows. This means that
+when we report a retention value for 2020-01-01, we're talking about what
+portion of clients active on 2020-01-01 are still active some number of days
+later.
+
+When we were talking about backward-looking windows, our anchor date or "day 0"
+was always the `submission_date`. When we define forward-looking windows, however,
+we always choose an anchor date some time in the past. How we number the individual
+bits depends on what anchor date we choose.
+
+For example, in [GUD](../tools/gud.md), we show a "1-Week Retention" which considers a window of 14 days.
+For each client active in "week 0" (days 0 through 6), we determine retention by
+checking if they were also active in "week 1" (days 0 through 13).
+
+To make "1-Week Retention" more concrete,
+let's consider the same client as before, grabbing the `days_seen_bits` value from
+`clients_last_seen` with `submission_date = 2020-01-28`. We count back 13 bits in
+the array to define our new "day 0" which corresponds to 2020-01-15:
+
+```
+    0  0  0  0  0  0  0  0  0  0  0  0  0  0  1  0  0  0  0  0  0  1  0  0  0  0  0  0
+    ──────────────────────────────────────────────────────────────────────────────────
+                                              │  │  │  │  │  │  │  │  │  │  │  │  │  │
+                                              │  1  │  3  │  5  │  7  │  9  │ 11  │ 13
+                                              0     2     4     6     8    10    12                                     
+                                            └────────────────────┘
+                                                    Week 0       └───────────────────┘
+                                                                        Week 1
+```
+
+This client has a bit set in both week 0 and in week 1, so logically this client
+can be considered retained. They should be counted in both the denominator and
+in the numerator for the "1-Week Retention" value on 2020-01-15.
+
+Extracting the bits for a specific week can be achieved via UDF:
+
+```
+SELECT
+  -- Signature is bits28_range(offset_to_day_0, start_bit, number_of_bits)
+  udf.bits28_range(days_seen_bits, 13, 0, 7) AS week_0_bits,
+  udf.bits28_range(days_seen_bits, 13, 7, 7) AS week_1_bits
+FROM
+  telemetry.clients_last_seen
+```
+
+And then we can turn those bits into a boolean indicating whether the client
+was active or not as:
+
+```
+SELECT
+  BIT_COUNT(udf.bits28_range(days_seen_bits, 13, 0, 7)) > 0 AS active_in_week_0
+  BIT_COUNT(udf.bits28_range(days_seen_bits, 13, 7, 7)) > 0 AS active_in_week_1
+FROM
+  telemetry.clients_last_seen
+```
+
+The `bits28_range` function is powerful and flexible, but a bit cumbersome.
+We provide a `bits28_retention` convenience function that returns a nested
+structure with the standard retention definitions precalculated:
+
+```
+SELECT
+  submission_date,
+  udf.bits28_retention(days_seen_bits, submission_date).day_13.*
```

Comment:
I'd appreciate thoughts on the interface for this `bits28_retention` function. I wanted to make it easy to use and in particular wanted to limit the number of parameters that you have to pass in. The idea of having this return a whole struct of the various retention metrics is a bit weird, but hopefully is convenient?

Is it clear from the description how to use this?